### DISCUSSION TITLE
Possibly Working on RPi 4B!

### DISCUSSION THREAD

**Discussion by cjbell630** [thefloppydriver/pc2drc#14] [Show and tell]
So the README said RPi 4 support was in the future, but I was bored so I tried to see if I could get it to work. And I think I did! The module was installed, but I don't have a compatible wifi adapter - yet. I'll update this when I get one. Until then, here's what I did to at least get to the "Installed successfully, but no devices were using mac80211" point:

#### Note: 

I got basically all of it from [this](https://eev.ee/blog/2015/11/28/zdoom-on-wii-u-gamepad-with-raspberry-pi/) guide, which I consulted after like an hour or two of trying to figure out how to get the source for my exact rpi kernel. Idk how much of it is really necessary, and to me it looks like it could be cleaned up some, but I didn't want to change any of it until after I was sure it worked. Once I get it connected to the Gamepad and everything, I'll probably go back and make it look nicer :)

### Steps (so far):

1. Run stage 0 and stage 1
2. Stage 1 will ask you to download kernel sources manually. Open a new terminal and cd to the directory shown by stage 1, then run the following: (commands use sudo bc of directory perms)
```shell
firmware_hash=$(zcat /usr/share/doc/raspberrypi-bootloader/changelog.Debian.gz | grep -Pom 1 '(?<=firmware as of )[0-9a-f]{6,}')
linux_hash=$(curl -L https://raw.github.com/raspberrypi/firmware/$firmware_hash/extra/git_hash)
sudo wget -O rpi-linux.tar.gz https://github.com/raspberrypi/linux/tarball/$linux_hash
sudo tar -zxf rpi-linux.tar.gz
```
3. The source for the version of the kernel being used has been downloaded and unzipped. (untarred?) cd to the directory it just created (should be `rpi-linux-` followed by some characters) and run the following:
```shell
sudo modprobe configs
sudo zcat /proc/config.gz > .config
yes "" | sudo make oldconfig
sudo make modules_prepare
sudo wget https://github.com/raspberrypi/firmware/raw/$firmware_hash/extra/Module.symvers
```
4. On the terminal running stage 1, hit enter to continue.

You should eventually see "Woohoo! The module installed successfully!" :D

The rest of the setup might not work even if you have a compatible wifi adapter, idk. But I figured I'd put my steps here for now, before I forget lol. Lmk if anyone else gets it fully running before I do!!

**Comment by thefloppydriver** [comment]
Good job! I'll test it out on my pi when I get the time, I have a lot of bugs to fix first though :)
